cc-check_test.go

```diff
@@ -472,80 +539,118 @@ func TestCheckHostIsClearContainersCapable(t *testing.T) {
 		{"GenuineIntel", "lm vmx sse4_1", false},
 	}
 
+	moduleData := []testModuleData{
+		{filepath.Join(sysModuleDir, "kvm"), true, ""},
+		{filepath.Join(sysModuleDir, "kvm_intel"), true, ""},
+		{filepath.Join(sysModuleDir, "kvm_intel/parameters/nested"), false, "Y"},
+		{filepath.Join(sysModuleDir, "kvm_intel/parameters/unrestricted_guest"), false, "Y"},
+	}
+
+	setupCheckHostIsClearContainersCapable(t, cpuInfoFile, cpuData, moduleData)
+
+	// remove the modules to force a failure
+	err = os.RemoveAll(sysModuleDir)
+	assert.NoError(t, err)
+
+	err = hostIsClearContainersCapable(cpuInfoFile)
+	assert.Error(t, err)
+}
+
+func TestCCCheckCLIFunction(t *testing.T) {
 	dir, err := ioutil.TempDir("", "")
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(dir)
 
-	file := filepath.Join(dir, "cpuinfo")
+	logfile := filepath.Join(dir, "global.log")
 
 	savedSysModuleDir := sysModuleDir
+	savedProcCPUInfo := procCPUInfo
+
+	cpuInfoFile := filepath.Join(dir, "cpuinfo")
 
 	// XXX: override
 	sysModuleDir = filepath.Join(dir, "sys/module")
+	procCPUInfo = cpuInfoFile
 
 	defer func() {
 		sysModuleDir = savedSysModuleDir
+		procCPUInfo = savedProcCPUInfo
 	}()
 
 	err = os.MkdirAll(sysModuleDir, testDirMode)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	cpuData := []testCPUData{
+		{"GenuineIntel", "lm vmx sse4_1", false},
+	}
+
 	moduleData := []testModuleData{
-		{filepath.Join(sysModuleDir, "kvm"), true, ""},
-		{filepath.Join(sysModuleDir, "kvm_intel"), true, ""},
-		{filepath.Join(sysModuleDir, "kvm_intel/parameters/nested"), false, "Y"},
 		{filepath.Join(sysModuleDir, "kvm_intel/parameters/unrestricted_guest"), false, "Y"},
+		{filepath.Join(sysModuleDir, "kvm_intel/parameters/nested"), false, "Y"},
```

Comment:
Good question. Technically not, but if it isn't available you won't be able to run a kvm-enabled VM inside a Clear Container (which might be surprising to users0. I guess strictly it should be a warning. #281 raised to deal with that scenario.